actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/BatchOperationNotifierActor.java

```diff
@@ -0,0 +1,335 @@
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.datasecurity.DecryptionService;
+import org.sunbird.common.models.util.mail.SendMail;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.actors.notificationservice.dao.EmailTemplateDao;
+import org.sunbird.learner.actors.notificationservice.dao.impl.EmailTemplateDaoImpl;
+import org.sunbird.learner.util.Util;
+import org.sunbird.models.course.batch.CourseBatch;
+
+/**
+ * This actor will handle course enrollment operations.
+ *
+ * @author github.com/iostream04
+ */
+
+/** @author mayank */
+@ActorConfig(
+  tasks = {"batchBulk", "batchUpdate", "batchOperation"},
+  asyncTasks = {"batchBulk", "batchUpdate", "batchOperation"}
+)
+public class BatchOperationNotifierActor extends BaseActor {
+  private static CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+
+  private DecryptionService decryptionService =
+      org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.getDecryptionServiceInstance(
+          null);
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    if (request.getOperation().equalsIgnoreCase(ActorOperations.BATCH_BULK.getValue())) {
+      Map<String, Object> requestMap = request.getRequest();
+      batchCreateOperationNotifier(
+          (CourseBatch) requestMap.get(BackgroundOperations.COURSE_BATCH.name()),
+          (String) requestMap.get(BackgroundOperations.OPERATION_TYPE.name()));
+    } else if (request.getOperation().equalsIgnoreCase(ActorOperations.BATCH_UPDATE.getValue())) {
+      Map<String, Object> requestMap = request.getRequest();
+      batchUpdateOperationNotifier(
+          (CourseBatch) requestMap.get(BackgroundOperations.OLD.name()),
+          (CourseBatch) requestMap.get(BackgroundOperations.NEW.name()));
+    } else if (request
+        .getOperation()
+        .equalsIgnoreCase(ActorOperations.BATCH_OPERATION.getValue())) {
+      Map<String, Object> requestMap = request.getRequest();
+      batchEnrollOperationNotifier(
+          (Map<String, Object>) requestMap.get(BackgroundOperations.COURSE_MAP.name()),
+          (CourseBatch) requestMap.get(BackgroundOperations.COURSE_BATCH.name()),
+          (String) requestMap.get(BackgroundOperations.OPERATION_TYPE.name()));
+    } else {
+      onReceiveUnsupportedOperation(request.getOperation());
+    }
+  }
+
+  /*
+   * This method takes courseBatch and operationType String as parameter and
+   * process invite-only batch enrol and unenroll operation notifications mentor
+   * and learner add remove Operation
+   */
+  private void batchCreateOperationNotifier(CourseBatch courseBatch, String operationType) {
+    if (operationType.equals(JsonKey.ADD)) {
```

Comment:
What will happen if operation type is neither ADD nor REMOVE. 
if it's always ADD or REMOVE then put only if  and else.

Reply:
done